source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    Cone3( const Line3<T>& inAxis, T inAngle, T inLength )
+        :
+        position( inAxis ),
+        angle( inAngle ),
+        length( inLength )
+    {}
+
+    inline MR::Vector3<T>& center( void )
+    {
+        return position.p;
+    }
+    inline MR::Vector3<T>& direction( void )
+    {
+        return position.d;
+    }
+    inline MR::Vector3<T>& apex( void )
+    {
+        return center();
+    }
+
+    MR::Line3<T> position; // the combination of the apex of the cone and the direction of its main axis in space
```

Comment:
direction is going down from apex or up from bse? (please make it clear in comment)

Reply:
better name it axis, `Line position` is confusing

Reply:
Line consists both apex and main line direction. 
But, ok. Will rename to axis